lib/cluster.js

```diff
@@ -260,6 +260,9 @@ function masterInit() {
       return process.nextTick(setupSettingsNT, settings);
     initialized = true;
     schedulingPolicy = cluster.schedulingPolicy;  // Freeze policy.
+    if (settings.scheduler) {
+      settings.scheduler(handles);
```

Comment:
Doesn't the scheduler function need to be called every time something is distributed?